Add command-line switch to omit folders from debug output

Fishbowler opened the pull request:
Continuing #1445 
Fixes #1427 

Adding a `--flatten-debug-output` switch that will remove the datetime folder, as well as `/.maestro/tests/` from the path, leaving the debug output exactly where the user defined.

(Updated with feedback from previous PR to remove 

## Proposed Changes

Added a new option `--flatten-debug-output` to the `test` command in `maestro-cli` to simplify the test output folder structure for CI environments. Updated the `TestDebugReporter` class to handle this option and generate the output path accordingly.

## Testing

I'm using a simple flow to force an error:
```
appId: com.example
---
- assertTrue: 
    condition: ${ 1 === 2 }
    label: 1 is the same as 2
```

No params:
```
 ⇒ ./maestro test scratch.yaml 
Running on emulator-5554      
                              
 ║                            
 ║  > Flow                    
 ║                            
 ║    ❌  Does 1 equal 2?     
 ║                            
                              
Assertion is false: false is true

==== Debug output (logs & screenshots) ====

/Users/danc/.maestro/tests/2024-05-17_150239
```

Just the debug option:
```
 ⇒ ./maestro test --debug-output /tmp/maestro scratch.yaml                    
Running on emulator-5554        
                                
 ║                              
 ║  > Flow                      
 ║                              
 ║    ❌  1 is the same as 2    
 ║                              
                                
Assertion is false: false is true

==== Debug output (logs & screenshots) ====

/tmp/maestro/.maestro/tests/2024-05-17_150426
```

With the new option:
```
 ⇒ ./maestro test --debug-output /tmp/maestro --flatten-debug-output  scratch.yaml
Running on emulator-5554        
                                
 ║                              
 ║  > Flow                      
 ║                              
 ║    ❌  1 is the same as 2    
 ║                              
                                
Assertion is false: false is true

==== Debug output (logs & screenshots) ====

/tmp/maestro
```

Full file listing of /tmp/maestro after above:
```
 ⇒ find /tmp/maestro 
/tmp/maestro
/tmp/maestro/screenshot-❌-1715954725865-(scratch.yaml).png
/tmp/maestro/commands-(scratch.yaml).json
/tmp/maestro/.maestro
/tmp/maestro/.maestro/tests
/tmp/maestro/.maestro/tests/2024-05-17_150426
/tmp/maestro/.maestro/tests/2024-05-17_150426/commands-(scratch.yaml).json
/tmp/maestro/.maestro/tests/2024-05-17_150426/maestro.log
/tmp/maestro/.maestro/tests/2024-05-17_150426/screenshot-❌-1715954675177-(scratch.yaml).png
/tmp/maestro/maestro.log
```

## Issues Fixed

#1427

felipevolpone commented on maestro-cli/src/main/java/maestro/cli/report/TestDebugReporter.kt:
```diff
@@ -125,10 +127,11 @@ object TestDebugReporter {
-        val dateFormatter = DateTimeFormatter.ofPattern(dateFormat)
-        val folderName = dateFormatter.format(LocalDateTime.now())
-        val debugOutput = Paths.get(debugOutputPathAsString ?: System.getProperty("user.home"), ".maestro", "tests", folderName)
+        val preamble = if(flattenDebugOutput) arrayOf("") else arrayOf(".maestro", "tests")
```
@Fishbowler Instead of having this if in every line, let's create a method that returns the `debugOutput` when `flatterDebugOutput` is `true` and another method for when it's `false`

felipevolpone commented on maestro-cli/src/main/java/maestro/cli/command/TestCommand.kt:
```diff
@@ -87,6 +87,12 @@ class TestCommand : Callable<Int> {
 
+    @Option(
+        names = ["--flatten-debug-output"],
+        description = ["Don't create subfolders or timestamps for each run. Useful for CI."]
```
Let's improve this description, something like:
```
All file outputs from the test case are created in the folder without subfolders or timestamps for each run. It can be used with --debug-output. Useful for CI
```

Fishbowler replied:
Far prefer your wording, switching it to a positive stance.

felipevolpone commented on maestro-cli/src/main/java/maestro/cli/command/TestCommand.kt:
```diff
@@ -161,7 +167,9 @@ class TestCommand : Callable<Int> {
                 )
 
-                TestDebugReporter.deleteOldFiles()
+                if(!flattenDebugOutput){
```
nit: the code format seems wrong, there is no space between `if` and the `(` and between `)` and the `{`.

Fishbowler commented:
Thanks for the thorough review! Will rebase and address.

Fishbowler commented:
I've pushed new changes and rerun the same tests as above - all three give the same results.

Rather than split into 2 separate methods, I've split off some of the noisier stuff and refactored what's there. I could've reverted the stuff I split off to the one-liner that existed for this previously, but I think that this is more comprehendable for ongoing maintenance.

felipevolpone commented:
@Fishbowler thanks for handling the comments and changing the code. The PR LGTM. Between today and tomorrow, I'll get the branch locally, test it and if it's all good, I'll merge it. Thanks again 🙌

felipevolpone commented:
Great work @Fishbowler. Thank you for your contribution
